sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.hpp

```diff
@@ -14,3 +14,20 @@
 ur_result_t enqueueEventsWait(ur_queue_handle_t CommandQueue, CUstream Stream,
                               uint32_t NumEventsInWaitList,
                               const ur_event_handle_t *EventWaitList);
+
+void guessLocalWorkSize(ur_device_handle_t Device, size_t *ThreadsPerBlock,
+                        const size_t *GlobalWorkSize, const uint32_t WorkDim,
+                        const size_t MaxThreadsPerBlock[3],
+                        ur_kernel_handle_t Kernel, uint32_t LocalSize);
+
+bool hasExceededMaxRegistersPerBlock(ur_device_handle_t Device,
+                                     ur_kernel_handle_t Kernel,
+                                     size_t BlockSize);
+
+ur_result_t
+setKernelParams(const ur_context_handle_t hContext,
+                const ur_device_handle_t hDevice, const uint32_t workDim,
+                const size_t *pGlobalWorkOffset, const size_t *pGlobalWorkSize,
+                const size_t *pLocalWorkSize, ur_kernel_handle_t &hKernel,
+                CUfunction &hCuFunc, size_t (&ThreadsPerBlock)[3],
+                size_t (&BlocksPerGrid)[3]);
```

Comment:
nitpick - don't need the `h`/`p` prefixes for this helper function (only the UR entry-points). Can use PascalCase like the functions above.